src/diart/blocks/diarization.py

```diff
@@ -21,7 +21,7 @@ class SpeakerDiarizationConfig(base.PipelineConfig):
     def __init__(
         self,
         segmentation: Optional[m.SegmentationModel] = None,
-        embedding: Optional[m.EmbeddingModel] = None,
+        embedding: Optional[Union[m.EmbeddingModel, m.WeSpeakerSpeakerEmbeddingInference]] = None,
```

Comment:
This approach complicates the way of adding new custom models. We need to make this transparent to the user and always keep `m.EmbeddingModel`. See my other comments for how to address this.